app/src/main/java/org/wikipedia/readinglist/AddToReadingListDialog.java

```diff
@@ -201,14 +205,33 @@ private void showCreateListDialog() {
     }
 
     private void addAndDismiss(final ReadingList readingList, final PageTitle title) {
-
         if (readingList.pages().size() >= SiteInfoClient.getMaxPagesPerReadingList()) {
             String message = getString(R.string.reading_list_article_limit_message, readingList.title(), SiteInfoClient.getMaxPagesPerReadingList());
             FeedbackUtil.makeSnackbar(getActivity(), message, FeedbackUtil.LENGTH_DEFAULT).show();
             dismiss();
             return;
         }
 
+        execute(readingList, title);
+    }
+
+    private void addAndDismiss(final ReadingList readingList, final List<PageTitle> titles) {
```

Comment:
Do we need a separate function for this? Can we have just a single `addAndDismiss` function that takes a list of Titles, and give it a singleton list in the case of a single title?

Reply:
Make sense, done!